### PR TITLE
refactor: remove project.GithubRepositoryURL in favour of project.GithubRepo

### PR DESCRIPTION
**Pull request by jan-zabloudil**

### CODE REVIEW

File: github/client.go [hunk 1]
```diff
@@ -71,19 +71,14 @@ func (c *Client) ReadRepo(ctx context.Context, tkn string, rawRepoURL string) (s
 	}, nil
 }
 
-func (c *Client) ReadTagsForRepo(ctx context.Context, tkn string, repoURL url.URL) ([]svcmodel.GitTag, error) {
-	repo, err := model.ToGithubRepo(repoURL)
-	if err != nil {
-		return nil, svcerrors.NewGithubRepoInvalidURL().Wrap(err).WithMessage(err.Error())
-	}
-
+func (c *Client) ReadTagsForRepo(ctx context.Context, tkn string, repo svcmodel.GithubRepo) ([]svcmodel.GitTag, error) {
```
Comment (jan-zabloudil) [inline github/client.go:74]: `ReadTagsForRepo` now also accepts `svcmodel.GithubRepo` instead of url

File: repository/project.go [hunk 2]
```diff
@@ -540,10 +538,9 @@ func toUpdateProjectArgs(p svcmodel.Project) pgx.NamedArgs {
 		"slackChannelID": p.SlackChannelID,
 		// Converted to the struct with json tags (the field is saved as json in the database).
 		"releaseNotificationConfig": model.ReleaseNotificationConfig(p.ReleaseNotificationConfig),
-		"githubRepositoryURL":       p.GithubRepoURL.String(), // TODO remove
-		"githubOwnerSlug":           ownerSlug,
-		"githubRepoSlug":            repoSlug,
-		"githubRepoURL":             repoURL.String(),
```
Comment (jan-zabloudil) [inline repository/project.go:546]: Here I am fixing one possible panic due to nil pointer

File: github/mock/client.go [hunk 3]
```diff
@@ -13,8 +13,8 @@ type Client struct {
 	mock.Mock
 }
 
-func (c *Client) ReadTagsForRepo(ctx context.Context, tkn string, repoURL url.URL) ([]svcmodel.GitTag, error) {
-	args := c.Called(ctx, tkn, repoURL)
+func (c *Client) ReadTagsForRepo(ctx context.Context, tkn string, repo svcmodel.GithubRepo) ([]svcmodel.GitTag, error) {
+	args := c.Called(ctx, tkn, repo)
```
Comment (bolshakoff) [inline github/mock/client.go:17]: Nice. With `repo` reads much smoother. ✨

File: repository/query/scripts/create_project.sql [hunk 4]
```diff
@@ -1,2 +1,2 @@
-INSERT INTO projects (id, name, slack_channel_id, release_notification_config, created_at, updated_at, github_repository_url)
-VALUES (@id, @name, @slackChannelID, @releaseNotificationConfig, @createdAt, @updatedAt, @githubRepositoryURL)
+INSERT INTO projects (id, name, slack_channel_id, release_notification_config, created_at, updated_at)
+VALUES (@id, @name, @slackChannelID, @releaseNotificationConfig, @createdAt, @updatedAt)
```
Comment (bolshakoff) [inline repository/query/scripts/create_project.sql:2]: OK.

**Review by bolshakoff — APPROVED** [review]

**Pull request merged**